examples/src/main/java/com/datastax/oss/driver/examples/mapper/killrvideo/video/CreateVideoQueryProvider.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.examples.mapper.killrvideo.video;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.BatchStatement;
+import com.datastax.oss.driver.api.core.cql.BatchStatementBuilder;
+import com.datastax.oss.driver.api.core.cql.BoundStatement;
+import com.datastax.oss.driver.api.core.cql.BoundStatementBuilder;
+import com.datastax.oss.driver.api.core.cql.DefaultBatchType;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.mapper.MapperContext;
+import com.datastax.oss.driver.api.mapper.entity.EntityHelper;
+import com.datastax.oss.driver.api.mapper.entity.saving.NullSavingStrategy;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.UUID;
+
+/**
+ * Provides the implementation of {@link VideoDao#create}.
+ *
+ * <p>Package-private visibility is sufficient, this will be called only from the generated DAO
+ * implementation.
+ */
+class CreateVideoQueryProvider {
+
+  private final CqlSession session;
+  private final EntityHelper<Video> videoHelper;
+  private final EntityHelper<UserVideo> userVideoHelper;
+  private final EntityHelper<LatestVideo> latestVideoHelper;
+  private final EntityHelper<VideoByTag> videoByTagHelper;
+  private final PreparedStatement preparedInsertVideo;
+  private final PreparedStatement preparedInsertUserVideo;
+  private final PreparedStatement preparedInsertLatestVideo;
+  private final PreparedStatement preparedInsertVideoByTag;
+
+  CreateVideoQueryProvider(
+      MapperContext context,
+      EntityHelper<Video> videoHelper,
+      EntityHelper<UserVideo> userVideoHelper,
+      EntityHelper<LatestVideo> latestVideoHelper,
+      EntityHelper<VideoByTag> videoByTagHelper) {
+    this.session = context.getSession();
+    this.videoHelper = videoHelper;
+    this.userVideoHelper = userVideoHelper;
+    this.latestVideoHelper = latestVideoHelper;
+    this.videoByTagHelper = videoByTagHelper;
+
+    this.preparedInsertVideo = prepareInsert(session, videoHelper);
+    this.preparedInsertUserVideo = prepareInsert(session, userVideoHelper);
+    this.preparedInsertLatestVideo = prepareInsert(session, latestVideoHelper);
+    this.preparedInsertVideoByTag = prepareInsert(session, videoByTagHelper);
+  }
+
+  void create(Video video) {
```

Comment:
In the `CreateUserQueryProvider.create` we have logic that is able to retry in a case of a failure.
Here we are trying only once and if the batch failed the exception is just propagated. In the first case, we are guarding against `UUID` collisions but I think that in this method `video.setVideoid(UUID.randomUUID());` can also lead to a collision. Shouldn't we handle that?

Reply:
I'm removing UUID collision handling for user creation per Alexandre's other comment.